core/src/test/java/io/questdb/test/cutlass/line/tcp/LineTcpBootstrapTest.java

```diff
@@ -44,6 +51,81 @@ public void setUp() {
         dbPath.parent().$();
     }
 
+    @Test
+    public void testAsciiFlagCorrectlySetForVarcharColumns() throws Exception {
+        TestUtils.assertMemoryLeak(() -> {
+            try (final ServerMain serverMain = new ServerMain(getServerMainArgs())) {
+                serverMain.start();
+                CairoEngine engine = serverMain.getEngine();
+                try (SqlExecutionContext sqlExecutionContext = TestUtils.createSqlExecutionCtx(engine)) {
+                    engine.ddl(
+                            "CREATE TABLE 'betfairRunners' (\n" +
+                                    "  id INT,\n" +
+                                    "  runner VARCHAR,\n" +
+                                    "  age BYTE,\n" +
+                                    "  remarks SYMBOL CAPACITY 2048 CACHE INDEX CAPACITY 256,\n" +
+                                    "  timestamp TIMESTAMP\n" +
+                                    ") timestamp (timestamp) PARTITION BY MONTH WAL\n" +
+                                    "DEDUP UPSERT KEYS(timestamp, id);",
+                            sqlExecutionContext
+                    );
+
+                    try (Sender sender = Sender.fromConfig("http::addr=localhost:" + serverMain.getHttpServerPort() + ";")) {
+                        for (int i = 0; i < 1; i++) {
+                            sender.table("betfairRunners").symbol("remarks", "SAw,CkdRnUp&2").
+                                    stringColumn("runner", "BallyMac Fifra")
+                                    .longColumn("id", 548738)
+                                    .longColumn("age", 58)
+                                    .at(TimestampFormatUtils.parseTimestamp("2024-06-30T00:00:00Z"), ChronoUnit.MICROS);
+                            sender.table("betfairRunners").symbol("remarks", "Fcd-Ck1").
+                                    stringColumn("runner", "BallyMac Fifra")
+                                    .longColumn("id", 548738)
+                                    .longColumn("age", 58)
+                                    .at(TimestampFormatUtils.parseTimestamp("2024-06-24T00:00:00Z"), ChronoUnit.MICROS);
+                            sender.table("betfairRunners").symbol("remarks", "(R8) LdRnIn військові").
+                                    stringColumn("runner", "BallyMac Fifra")
+                                    .longColumn("id", 548738)
+                                    .longColumn("age", 58)
+                                    .at(TimestampFormatUtils.parseTimestamp("2024-06-17T00:00:00Z"), ChronoUnit.MICROS);
+                            sender.flush();
+                        }
+                    }
+
+                    // server main already runs Apply2Wal job in the background. We have to wait for the table to catchup
```

Comment:
nit: we have `assertEventually` for this purpose.

Reply:
Let's do it in another PR